src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/ApiVersions.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Server
+{
+    public static class ApiVersions
+    {
+        public const string ParameterName = "api-version";
+
+        public const string May012017 = "2017-05-01.5.0";
+        public const string June012017 = "2017-06-01.5.1";
+        public const string Sept012017 = "2017-09-01.6.0";
+        public const string March012018 = "2018-03-01.6.1";
+        public const string August012018 = "2018-08-01.7.0";
+        public const string ApiVersionLatest = "9999-09-09.99.99";
+
+        private static readonly HashSet<string> ValidApiVersions = new HashSet<string>
```

Comment:
Perhaps change this to `ImmutableHashSet<string>` ?

Reply:
Done